12 try react redux

minump opened the pull request:
This PR is the redux implementation of [PR#20](https://github.com/clowder-framework/CONSORT-frontend/pull/20)
If needed, change the base branch from "main" to "19-add-a-overlay-when-processing-results" to see specific changes.

The rendering html content largely remains the same as in PR#20. Main differences will be incorporating redux, and implementation / re-arranging of some functions.

The create a dataset + upload file to dataset + submit for extraction is dispatched as one single action.
The overlay text (text inside the spinner wheel) changes according to processing status. The text changes from "Processing" to "Uploading file" to "Checking extraction status" to "Generating html". (although user might not be able to see "Processing" "uploading file" and "generating html" due to very short times for that specific tasks).

lmarini reviewed (changes requested):
Wroks well. I see the following error in the console when first loading the page. It might be because `filesInDataset` is not initialized yet.

"Uncaught (in promise) TypeError: filesInDataset[0] is undefined
    _callee2$ CreateAndUpload.js:49
    tryCatch CreateAndUpload.js:2
    makeInvokeMethod CreateAndUpload.js:2
    defineIteratorMethods CreateAndUpload.js:2
    asyncGeneratorStep CreateAndUpload.js:2
    _next CreateAndUpload.js:2
    _asyncToGenerator CreateAndUpload.js:2
    _asyncToGenerator CreateAndUpload.js:2
    React 5
    unstable_runWithPriority scheduler.development.js:468

minump commented:
> Wroks well. I see the following error in the console when first loading the page. It might be because `filesInDataset` is not initialized yet.
> 
> "Uncaught (in promise) TypeError: filesInDataset[0] is undefined _callee2$ CreateAndUpload.js:49 tryCatch CreateAndUpload.js:2 makeInvokeMethod CreateAndUpload.js:2 defineIteratorMethods CreateAndUpload.js:2 asyncGeneratorStep CreateAndUpload.js:2 _next CreateAndUpload.js:2 _asyncToGenerator CreateAndUpload.js:2 _asyncToGenerator CreateAndUpload.js:2 React 5 unstable_runWithPriority scheduler.development.js:468

Added if exists check on filesInDataset.
Updated code according to feedback. Ready for review. @lmarini

lmarini commented:
Sorry can you resolve conflicts @minump ? Thanks!
